src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseReport.java

```diff
@@ -0,0 +1,607 @@
+    thisMonthEighteen = getSingleTable(allPreciseSamples, allPreciseSamplesById, labelListShort,
+        TimePoint.predicate(TimePoint.EIGHTEEN), byCreatedBetween(start, end), bySampleCategory(SAMPLE_CATEGORY_TISSUE));
+    thisMonthTwentyFour = getSingleTable(allPreciseSamples, allPreciseSamplesById, labelListLong,
+        TimePoint.predicate(TimePoint.TWENTY_FOUR), byCreatedBetween(start, end), bySampleCategory(SAMPLE_CATEGORY_TISSUE));
+    thisMonthSlides = getSingleTable(allPreciseSamples, allPreciseSamplesById, labelListSlides, byCreatedBetween(start, end),
+        bySampleCategory(SAMPLE_CATEGORY_TISSUE_PROCESSING));
+
+    inventoryAvailable = getSingleTable(allPreciseSamples, allPreciseSamplesById, labelListInventory, byNotEmpty);
+
+    barcodeAndNameCodingMismatch = filter(allPreciseSamples, byBarcodeTimesReceivedMismatch);
+    slidesWithoutSlideTimepoint = filter(allPreciseSamples, byTimePointCodingMismatch);
+
+  }
+
+  private final Predicate<SampleDto> byBarcodeTimesReceivedMismatch = dto -> {
+    // check only tissues and slides with non-null barcodes
+    if ((!bySampleCategory(SAMPLE_CATEGORY_TISSUE).test(dto) && !bySampleCategory(SAMPLE_CATEGORY_TISSUE_PROCESSING).test(dto))
+        || dto.getTubeBarcode() == null) {
+      return false;
+    }
+
+    String barcodeString = dto.getTubeBarcode().split("\\.")[1].replaceFirst("^0", "");
+    String timesReceived = "";
+    if (bySampleCategory(SAMPLE_CATEGORY_TISSUE).test(dto)) {
+      timesReceived = dto.getName().split("-")[dto.getName().split("-").length - 1];
+    } else if (getAttribute(ATTR_SLIDES, dto) != null) {
+      // quick and dirty check for Slides
+      Matcher m = Pattern.compile(".*(\\d{1,5}-\\d{1,5}).*").matcher(dto.getName());
+      if (!m.matches()) throw new IllegalArgumentException("Sample with alias " + dto.getName() + " is malformed.");
+      timesReceived = m.group(1).split("-")[1];
+    } else {
+      return false;
+    }
+    return Integer.valueOf(barcodeString) != Integer.valueOf(timesReceived);
+  };
+
+  private final Predicate<SampleDto> byTimePointCodingMismatch = dto -> {
+    // finds slides that are not coded as biopsy (6) or radical prostatectomy (7) in the sample alias
+    if (getAttribute(ATTR_SLIDES, dto) == null) return false; // quick and dirty check for slides
+    Matcher m = Pattern.compile(".*(\\d{1,5}-\\d{1,5}).*").matcher(dto.getName());
+    if (!m.matches()) throw new IllegalArgumentException("Sample with alias " + dto.getName() + " is malformed.");
+    String timePoint = m.group(1).split("-")[0];
+    return Integer.valueOf(timePoint) < 6;
+  };
+
+  private final Predicate<SampleDto> byNotEmpty = sample -> {
+    if (sample.getSampleType() == null)
+      throw new IllegalArgumentException(String.format("Got very weird sample type for sample %s", sample.getName()));
+    if (sample.getSampleType().contains("Illumina") || "Unknown".equals(sample.getSampleType())) return false;
+    switch (sample.getSampleType()) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)